### PR TITLE
raftstore: simplify casual messages

### PR DESCRIPTION
**Pull request by hicqu**
Signed-off-by: qupeng <qupeng@pingcap.com>

### What problem does this PR solve?
Simplify casual message to make it easiler to migrate to raft engine

### Check List 

Tests 
- Unit test

Side effects

### Release note 
* No release note

### CODE REVIEW

File: components/raftstore/src/store/peer.rs [hunk 1]
```diff
@@ -3115,6 +3115,41 @@ fn make_transfer_leader_response() -> RaftCmdResponse {
     resp
 }
 
+/// A poor version of `Peer` to avoid port generic variables everywhere.
+pub trait PlainPeer {
```
Comment (BusyJay) [inline components/raftstore/src/store/peer.rs:3119]: What does `Plain` mean?

Comment (hicqu) [reply components/raftstore/src/store/peer.rs:3119]: For the context do you have some suggestions?

**Pull request merged**